rs/canister_sandbox/src/replica_controller/sandbox_process_eviction.rs

```diff
@@ -30,26 +46,27 @@ pub(crate) fn evict(
     max_count_threshold: usize,
     last_used_threshold: Instant,
 ) -> Vec<EvictionCandidate> {
-    candidates.sort_by_key(|x| x.last_used);
+    candidates.sort();
 
     let evict_at_least = candidates.len().saturating_sub(max_count_threshold);
     let evict_at_most = candidates.len().saturating_sub(min_count_threshold);
 
-    let mut evicted = vec![];
+    // candidates = candidates [0.. evict_at_least), remaining_candidates = candidates [evict_at_least.. candidates.len())
+    let remaining_candidates = candidates.split_off(evict_at_least);
+    let mut evicted = candidates;
 
-    for candidate in candidates.into_iter() {
+    for candidate in remaining_candidates.into_iter() {
```

Comment:
I guess now we could be iterating through all the canisters in more cases (if we don't hit `evict_at_most`). I guess that's fine since the list is short and this only runs every 10 seconds or so - agree?

Reply:
I think we might need to change the 10s logic to something smaller, but I'm not super sure yet, let's keep this under discussion.